subsys/bluetooth/host/conn.c

```diff
@@ -1220,7 +1235,17 @@ static struct bt_conn_tx *add_pending_tx(struct bt_conn *conn,
 
 	BT_DBG("conn %p cb %p user_data %p", conn, cb, user_data);
 
-	tx = k_fifo_get(&free_tx, K_FOREVER);
+	if (IS_ENABLED(CONFIG_BT_DEBUG_CONN)) {
+		tx = k_fifo_get(&free_tx, K_NO_WAIT);
+		if (!tx) {
+			BT_WARN("Unable to get a free conn_tx, yielding...");
+			k_yield();
+			tx = k_fifo_get(&free_tx, K_FOREVER);
```

Comment:
Doesn't calling k_fifo_get with a non-zero timeout (or K_FOREVER) yield implicitly, i.e. the k_yield() seems unnecessary.

Reply:
Yep, will remove it.

Reply:
Fixed